vstools/functions/render.py

```diff
@@ -57,6 +57,31 @@ def clip_async_render(
     callback: Callable[[int, vs.VideoFrame], T] | None = None, prefetch: int = 0,
     backlog: int = -1, y4m: bool | None = None, async_requests: int | bool | AsyncRenderConf = False
 ) -> list[T] | None:
+    """
+    Iterate over an entire clip and optionally write results in a file.
+
+    This is mostly useful for metric gathering that must be performed before you do anything else.
+    This could be for example gathering scenechanges, per-frame heuristics, etc.
+
+    It's highly recommended to perform as little filtering as possible on the input clip for speed purposes.
+
+    :param clip:            Clip to render.
+    :param outfile:         Optional binary output to write. @@EXPAND@@
+    :param progress:        @@PLACEHOLDER@@
```

Comment:
add a string to print a progress bar
add a callback that takes the current and num frames to do your own progress code